pkg/webhooks/validation/controller/controller.go

```diff
@@ -274,10 +263,13 @@ func newController(
 func (c *Controller) Start(stop <-chan struct{}) {
 	c.stopCh = stop
 	go c.startFileWatcher(stop)
-	go c.sharedInformers.Start(stop)
-
-	for _, ready := range c.sharedInformers.WaitForCacheSync(stop) {
-		if !ready {
+	if !cache.WaitForCacheSync(stop, c.webhookInformer.Informer().HasSynced) {
+		log.Errorf("failed to wait for cache sync")
+		return
+	}
+	if c.endpointsInformer != nil {
```

Comment:
Would lines 270-275 go away once the endpoint shared informer is plumbed through to the webhook controller?

Reply:
I am not actually sure this is redundant. We still want to wait until its synced before we start the queue?

Reply:
We want to wait. Should that be the responsibility of each controller using the shared informers, or the top-level code that is starting the informers?

Reply:
I would say each controller. My reasoning is k8s does it this way (https://github.com/kubernetes/kubernetes/blob/94e5537fa0973f6f8a252635592b43dc65f31126/pkg/controller/endpointslice/endpointslice_controller.go#L240) and we would need to be very careful that we call the code to wait prior to starting any controllers.

Reply:
Sounds reasonable.  So the way to look at this is the WaitForCacheSync is the redundant call, possibly because we haven't updated individual callers yet to wait themselves?